File: dart/lib/model/v3_bulk_departures_route_direction_response.dart
part of swagger.api;

class V3BulkDeparturesRouteDirectionResponse {
  /* Route identifier */
  String routeId = null;
/* Direction of travel identifier */
  int directionId = null;
/* Name of direction of travel */
  String directionName = null;

  V3BulkDeparturesRouteDirectionResponse();

  @override
  String toString() {
    return 'V3BulkDeparturesRouteDirectionResponse[routeId=$routeId, directionId=$directionId, directionName=$directionName, ]';
  }

  V3BulkDeparturesRouteDirectionResponse.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    routeId = json['route_id'];
    directionId = json['direction_id'];
    directionName = json['direction_name'];
  }

  Map<String, dynamic> toJson() {
    return {
      'route_id': routeId,
      'direction_id': directionId,
      'direction_name': directionName
     };
  }

  static List<V3BulkDeparturesRouteDirectionResponse> listFromJson(List<dynamic> json) {
    return json == null ? new List<V3BulkDeparturesRouteDirectionResponse>() : json.map((value) => new V3BulkDeparturesRouteDirectionResponse.fromJson(value)).toList();
  }

  static Map<String, V3BulkDeparturesRouteDirectionResponse> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, V3BulkDeparturesRouteDirectionResponse>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new V3BulkDeparturesRouteDirectionResponse.fromJson(value));
    }
    return map;
  }
}

File: csharp-dotnet2/src/main/CsharpDotNet2/IO/Swagger/Model/V3BulkDeparturesRouteDirectionResponse.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class V3BulkDeparturesRouteDirectionResponse {
    /// <summary>
    /// Route identifier
    /// </summary>
    /// <value>Route identifier</value>
    [DataMember(Name="route_id", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "route_id")]
    public string RouteId { get; set; }

    /// <summary>
    /// Direction of travel identifier
    /// </summary>
    /// <value>Direction of travel identifier</value>
    [DataMember(Name="direction_id", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "direction_id")]
    public int? DirectionId { get; set; }

    /// <summary>
    /// Name of direction of travel
    /// </summary>
    /// <value>Name of direction of travel</value>
    [DataMember(Name="direction_name", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "direction_name")]
    public string DirectionName { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class V3BulkDeparturesRouteDirectionResponse {\n");
      sb.Append("  RouteId: ").Append(RouteId).Append("\n");
      sb.Append("  DirectionId: ").Append(DirectionId).Append("\n");
      sb.Append("  DirectionName: ").Append(DirectionName).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}
